aten/src/ATen/Context.cpp

```diff
@@ -78,6 +78,36 @@ void Context::alertNotDeterministic(c10::string_view const& caller) {
   }
 }
 
+void Context::alertCuBLASConfigNotDeterministic() {
+  if (deterministic()) {
+    static const char config_var_name[] = "CUBLAS_WORKSPACE_CONFIG";
+    static const char config0[] = ":4096:8";
+    static const char config1[] = ":16:8";
+    if (!cublas_config_captured) {
```

Comment:
Don't do it this way, you've introduced a data race if multiple threads concurrently call this function.

A safe and convenient way to do the initialization once idiom is to simply define a static variable in your function

```
static bool cublas_config_deterministic = ...
```

Reply:
Thanks, didn't realize that would be a thread safety issue.

Reply:
The new code still isn't thread safe ;) Go read https://stackoverflow.com/questions/8102125/is-local-static-variable-initialization-thread-safe-in-c11

The *initialization* of the static variable (`...` in my example) is thread safe, but NOT subsequent accesses to the variable.

Reply:
Oh, sorry, I misread your code snippet, thought you were talking about the initialization of `cublas_config_captured`. I understand now, I'll fix it.